packages/mobx-react/src/disposeOnUnmount.ts

```diff
@@ -17,12 +20,22 @@ function runDisposersOnWillUnmount() {
     })
 }
 
+/**
+ * @deprecated
+ */
 export function disposeOnUnmount(target: React.Component<any, any>, propertyKey: PropertyKey): void
+
+/**
+ * @deprecated
+ */
 export function disposeOnUnmount<TF extends Disposer | Array<Disposer>>(
     target: React.Component<any, any>,
     fn: TF
 ): TF
 
+/**
+ * @deprecated
```

Comment:
NIT: Let's clarify why it is deprecated, as that shows up in VSCode suggestions

Reply:
Done, but message probably isn't entirely accurate. I am no sure which react version started to perform re-mounting. Maybe it's already done by 17 only in StrictMode, but I don't remember.